alf/optimizers/traj_optimizers.py

```diff
@@ -43,22 +63,19 @@ def __init__(self,
         self._upper_bound = upper_bound
         self._lower_bound = lower_bound
 
-    def obtain_solution(self, time_step: ActionTimeStep, state):
+    def obtain_solution(self, time_step: TimeStep, state):
         """Minimize the cost function provided
 
         Args:
-            time_step (ActionTimeStep): the initial time_step to start rollout
+            time_step (TimeStep): the initial time_step to start rollout
             state: input state to start rollout
         """
         init_obs = time_step.observation
         batch_size = init_obs.shape[0]
-        solutions = tf.random.uniform(
-            [batch_size, self._population_size, self._solution_dim],
-            self._lower_bound, self._upper_bound)
+        solutions = torch.rand(
+            batch_size, self._population_size, self._solution_dim
+        ) * (self._upper_bound - self._lower_bound) + self._lower_bound * 1.0
         costs = self.cost_function(time_step, state, solutions)
-        min_ind = tf.cast(tf.argmin(costs, axis=-1), tf.int32)
-        population_ind = tf.expand_dims(min_ind, 1)
-        batch_ind = tf.expand_dims(tf.range(tf.shape(solutions)[0]), 1)
-        ind = tf.concat([batch_ind, population_ind], axis=1)
-        solution = tf.gather_nd(solutions, ind)
+        min_ind = torch.argmin(costs, dim=-1).long()
+        solution = solutions.index_select(1, min_ind).squeeze(1)
```

Comment:
Is `squeeze(1)` necessary?

Reply:
It is necessary to remove the singular population dim. 
Can possibly use view/reshape as well. Added a comment on the change of tensor shapes for better documentation.

Reply:
The examples given by pytorch index_select does not suggest having a singular dimension.
Actually, the shape of `solutions.index_select(1, min_ind)` is [B, B, sol_dim].